server/src/main/java/org/candlepin/policy/entitlement/EntitlementRulesTranslator.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.policy.entitlement;
+
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.Pool;
+import org.candlepin.model.Product;
+import org.candlepin.policy.RulesValidationError;
+import org.candlepin.policy.RulesValidationWarning;
+import org.xnap.commons.i18n.I18n;
+
+import com.google.inject.Inject;
+
+
+/**
+ * Translates error keys returned from entitlement rules into translated error messages.
+ */
+public class EntitlementRulesTranslator {
+
+    // TODO:
+    // Add documentation for the intended circumstances for these error keys
+
+    /**
+     * Keys that represent the various warnings that can be produced by the EntitlementRules.
+     * Warning messages may use the following variable(s):
+     * {0} - pool id
+     */
+    public enum WarningKeys implements RulesValidationWarning {
+        ARCHITECTURE_MISMATCH(
+            "The entitlement's product architecture does not match with the consumer's."),
```

Comment:
You'll need to mark these with ```I18n.marktr``` so the translation stuff can pick it up. Otherwise the strings will be passthrough when i18n.tr is called and won't ever get translated.

Reply:
Done